android/app/src/main/java/com/microsoft/reacttestapp/Session.kt

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.reacttestapp
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import androidx.core.content.edit
+
+class Session(context: Context) {
+
+    private companion object {
+        const val MANIFEST_CHECKSUM = "ManifestChecksum"
+        const val REMEMBER_LAST_COMPONENT_ENABLED = "RememberLastComponent/Enabled"
+        const val REMEMBER_LAST_COMPONENT_INDEX = "RememberLastComponent/Index"
+    }
+
+    private val sharedPreferences: SharedPreferences =
+        context.getSharedPreferences(BuildConfig.APPLICATION_ID, MODE_PRIVATE)
+
+    private var lastComponentIndex: Int
+        get() = sharedPreferences.getInt(REMEMBER_LAST_COMPONENT_INDEX, -1)
+        set(index) = sharedPreferences.edit { putInt(REMEMBER_LAST_COMPONENT_INDEX, index) }
```

Comment:
After edit, you need to commit changes, ref. https://developer.android.com/reference/android/content/SharedPreferences.Editor.

Same for the other edits below

Reply:
We're using the [`SharedPreferences.edit` extension](https://developer.android.com/reference/kotlin/androidx/core/content/package-summary#edit) which should automatically call [`apply`](https://developer.android.com/reference/android/content/SharedPreferences/Editor.html#apply()) when the block exits.